dm/pkg/ha/stage.go

```diff
@@ -172,25 +176,34 @@ func GetAllRelayStage(cli *clientv3.Client) (map[string]Stage, int64, error) {
 // if task name is "", it will return all subtasks' stage as a map{task-name: stage} for the source.
 // if task name is given, it will return a map{task-name: stage} whose length is 1.
 func GetSubTaskStage(cli *clientv3.Client, source, task string) (map[string]Stage, int64, error) {
+	return getStageByKey(cli, common.StageSubTaskKeyAdapter, source, task, 0)
+}
+
+func getStageByKey(cli *clientv3.Client, key common.KeyAdapter, source, task string, revision int64) (map[string]Stage, int64, error) {
 	ctx, cancel := context.WithTimeout(cli.Ctx(), etcdutil.DefaultRequestTimeout)
 	defer cancel()
 
 	var (
 		stm  = make(map[string]Stage)
 		resp *clientv3.GetResponse
 		err  error
+		opts = make([]clientv3.OpOption, 0)
 	)
+	if revision > 0 {
```

Comment:
can we always use the latest revision? in future we might decouple from etcd, and revision may not be supported in new metadata storage.

Reply:
in `operateSubTaskStage`, we need to get validator stage using revision of the subtask.

although is't right to use latest revision(same result), but it's more clear this way.